### DISCUSSION TITLE
Find out whether fetchPreviousPage or fetchNextPage caused the error using a bi-directional useInfiniteQuery

### DISCUSSION THREAD

**Discussion by suzaninfi** [TanStack/query#3880] [Q&A]
When you're using a bi-directional `useInfiniteQuery`, and an error occurs while fetching, is there any way to know whether that error was caused by `fetchPreviousPage` or `fetchNextPage` (or neither in case of initial load)?

For example, I'm using 'previous' and 'next' buttons to fetch the previous or next page, and would like to hide the 'previous' button when an error occured while fetching the previous page (but not the 'next' button).

One of my ideas was to give an extra argument to `fetchNextPage` or `fetchPreviousPage`, to tell the query function which of these was used so it can throw a custom error. But I was wondering if there was a simpler way. In addition, I'm not sure how to add extra arguments to these without overwriting the `pageParam`.

My current code looks something like this, calling `fetchNextPage` and `fetchPreviousPage` on the buttons.

```ts
  const {
    data,
    error,
    fetchNextPage,
    fetchPreviousPage,
  } = useInfiniteQuery<{journeys: Journey[]; offset: number}, Error>(
    ['journeys', JSON.stringify(settings)],
    (data) => fetchJourneys(data.pageParam || 0, settings),
    {
      getNextPageParam: (lastPage) => {
        return lastPage.offset + 1;
      },
      getPreviousPageParam: (lastPage) => {
        return lastPage.offset - 1;
      },
    },
  );
```

Thank you! 🙏

**Comment by TkDodo** [comment]
not that I know of. We do have this information though because we are computing `isFetchingNextPage` from it:

https://github.com/TanStack/query/blob/357ec041a6fcc4a550f3df02c12ecc7bcdefbc05/packages/query-core/src/infiniteQueryObserver.ts#L151-L156

question is: what would be a good way to expose it 🤔

**Reply by suzaninfi** [reply]
Hm, maybe it would be useful to have access to the fetch direction? Not sure if that has downsides. Guess it's only relevant if you're working bidirectionally

**Reply by TkDodo** [reply]
thought so, too, but it would have to be `lastFetchDirection` or something ...
